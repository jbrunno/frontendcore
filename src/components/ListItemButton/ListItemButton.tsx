import React from 'react';
import {
  ListItemButton as MuiListItemButton,
  ListItemButtonProps as MuiListItemButtonProps,
} from '@mui/material';

export type ListItemButtonProps = MuiListItemButtonProps;

export function ListItemButton(props: ListItemButtonProps) {
  return <MuiListItemButton {...props} />;
}
export default ListItemButton;
