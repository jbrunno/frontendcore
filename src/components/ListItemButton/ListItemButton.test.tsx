import React from 'react';
import { render } from '@test-utils/component-testing';
import ListItemButton from './ListItemButton';

test('renders the ListItemButton', () => {
  const { container } = render(<ListItemButton />);
  expect(container).toMatchSnapshot();
});
