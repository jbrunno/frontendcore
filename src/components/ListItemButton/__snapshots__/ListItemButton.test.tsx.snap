// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ListItemButton 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-width: 0;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0000000A;
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-selected {
  background-color: rgba(74, 49, 255, 0.08);
}

.emotion-0.Mui-selected.Mui-focusVisible {
  background-color: rgba(74, 49, 255, 0.2);
}

.emotion-0.Mui-selected:hover {
  background-color: rgba(74, 49, 255, 0.12);
}

@media (hover: none) {
  .emotion-0.Mui-selected:hover {
    background-color: rgba(74, 49, 255, 0.08);
  }
}

.emotion-0.Mui-focusVisible {
  background-color: #0000001F;
}

.emotion-0.Mui-disabled {
  opacity: 0.38;
}

.emotion-1 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters emotion-0"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiTouchRipple-root emotion-1"
    />
  </div>
</div>
`;
